riscv/csrs.h

```diff
@@ -122,6 +122,23 @@ class pmpcfg_csr_t: public csr_t {
   virtual bool unlogged_write(const reg_t val) noexcept override;
 };
 
+class mseccfg_csr_t: public csr_t {
+  public:
+    mseccfg_csr_t(processor_t* const proc, const reg_t addr);
+    virtual reg_t read() const noexcept override;
+    bool get_mml() const noexcept;
+    bool get_mmwp() const noexcept;
+    bool get_rlb() const noexcept; 
+  protected:
+    virtual bool unlogged_write(const reg_t val) noexcept override;
+  private:
+    reg_t mseccfg_val;
+    reg_t pmplock_recorded;
+    friend class pmpcfg_csr_t; //pmpcfg needs to access pmplock_recorded
```

Comment:
Instead of `friend` how about a simple method: `record_lock(size_t idx, bool locked)`? But this is moot since I don't think the lock bits should be recorded in this class in the first place. (See comment on `pmplock_recorded`)

Reply:
Done. See 2nd commit.